npm/private/utils.bzl

```diff
@@ -156,6 +182,115 @@ def _npm_registry_download_url(package, version, registries, default_registry):
 def _is_git_repository_url(url):
     return url.startswith("git+ssh://")
 
+def _to_registry_url(url):
+    return "{}://{}".format(DEFAULT_REGISTRY_PROTOCOL, url) if url.find("//") == -1 else url
+
+def _default_registry():
+    return _to_registry_url("registry.npmjs.org/")
+
+def _hash(s):
+    result = str(hash(s))
+    if result.startswith("-"):
```

Comment:
meh, this seems like more code for us to maintain that's not really worth it just because you don't particularly enjoy the negative numbers. Is there any benefit to them all being positive?

Reply:
just to keep the hashes to `[a-zA-Z0-9]`. negative numbers are surprising hash values but then so are integers.

I spend a whole 10 minutes on this function so I could go either way

Reply:
I think we should just allow negative. Maybe someone will have a reason to produce this file themselves. Simpler to use the value starlark provides

Reply:
Ahh there it is: https://github.com/bazelbuild/starlark/issues/36#issuecomment-1115352085

Reply:
that's a good point; it may be a dumb looking hash but you can generate the same value with other languages